server/schedule/checker/rule_checker.go

```diff
@@ -152,7 +155,19 @@ func (c *RuleChecker) replaceRulePeer(region *core.RegionInfo, rf *placement.Rul
 		return nil, errors.New("no store to replace peer")
 	}
 	newPeer := &metapb.Peer{StoreId: store, Role: rf.Rule.Role.MetaPeerRole()}
-	op, err := operator.CreateMovePeerOperator("replace-rule-"+status+"-peer", c.cluster, region, operator.OpReplica, peer.StoreId, newPeer)
+	// randomly pick leader to avoid the Offline store be snapshot generotor bottleneck.
+	var newLeader *metapb.Peer
+	if region.GetLeader().GetId() == peer.GetId() {
```

Comment:
We expect the whole cluster to share the pressure of the downline as much as possible, 1/3 of the downline nodes are theoretically leaders, and then 1/3 of these leaders remain unchanged, so there is actually a steady 1/9 of the pressure concentrated on the downline node